osu.Game/Screens/Select/Carousel/CarouselGroup.cs

```diff
@@ -86,13 +87,15 @@ public override void Filter(FilterCriteria criteria)
 
             items.ForEach(c => c.Filter(criteria));
 
+            var sortByDateAdded = new FilterCriteria { Sort = SortMode.DateAdded };
+
             criteriaComparer = Comparer<CarouselItem>.Create((x, y) =>
             {
                 int comparison = x.CompareTo(criteria, y);
                 if (comparison != 0)
                     return comparison;
 
-                return x.ItemID.CompareTo(y.ItemID);
+                return -x.CompareTo(sortByDateAdded, y);
```

Comment:
This needs an inline explanation of what it's doing, and why `ItemID` can't be used. `ItemID`'s xmldoc describes it as the default sort, which you've now changed, so you'd also need to update or potentially remove that if going in this direction.

Reply:
We've been through this but the [relevant review thread](https://github.com/ppy/osu/pull/21927#discussion_r1059616670) has been collapsed.

Reply:
I saw that, but I think the points I make still apply. `ItemID` can be removed if `DateAdded` is to be used instead (although do note that date added has only a per-second precision and may lead to unstable sorting as a result...).

Reply:
> I saw that, but I think the points I make still apply. `ItemID` can be removed if `DateAdded` is to be used instead

Fair call.

> although do note that date added has only a per-second precision and may lead to unstable sorting as a result...

In normal usage this shouldn't matter but I guess it might when importing. What even is the option then? We don't have sequential database IDs anymore, so lexicographical sorting via realm GUIDs maybe?

Reply:
> so lexicographical sorting via realm GUIDs maybe?

May be best, yeah.

Reply:
Coming back from https://github.com/ppy/osu/pull/21927#discussion_r1059592231, I would hope for a `ReplaceItem` method which would preserve the ID of the item rather than allocating a new one, so that `UpdateBeatmapSet` would use it instead of removing and adding one. Failing that, using realm GUIDs seems like a good option.

Reply:
Adding `ReplaceItem()` is piling stuff on top of stuff. Using a better ordering method that allows to _get rid of_ `ItemID` is removing stuff. The second seems preferable to me.